azure_monitor/src/azure_monitor/export/metrics/__init__.py

```diff
@@ -52,21 +54,35 @@ def metric_to_envelope(
 
         if not metric_record:
             return None
+        # TODO: Opentelemetry does not have last updated timestamp for observer
+        # type metrics yet.
+        _time = time_ns()
```

Comment:
We may want to track this as an issue in this repo to be sure time is more accurate as soon is available in OT

Reply:
https://github.com/microsoft/opentelemetry-exporters-python/issues/59